### DISCUSSION TITLE
oxfordjournals.org feeds cannot be added

### DISCUSSION THREAD

**Discussion by svigneau** [FreshRSS/FreshRSS#4746] [General]
The feed `http://nar.oxfordjournals.org/rss/current.xml` cannot be added with the log message: `https://academic.oup.com/rss/site_5127/3091.xml; the status code is 200 and content-type is text/html [https://academic.oup.com/rss/site_5127/3091.xml]`. The corresponding feed validation service assessment is [here](https://validator.w3.org/feed/check.cgi?url=http%3A%2F%2Fnar.oxfordjournals.org%2Frss%2Fcurrent.xml) and indicates that it is a valid RSS feed but with some issues. 
The same applies to all RSS feeds from `oxfordjournals.org`, an important science journal publisher. However, other readers (e.g. Feedly) work with feeds from this publisher.
Is there a way to change FreshRSS settings to be able to read feeds from oxfordjournals.org despite imperfect formatting?

**Comment by aledeg** [comment]
Have you tried to use the `#force` option when registering the feed?

**Reply by svigneau** [reply]
Thank you for the suggestion! I tried to add `http://nar.oxfordjournals.org/rss/current.xml#force_feed`, but this fails too with the log message `Unknown error for feed [http://nar.oxfordjournals.org/rss/current.xml#force_feed]`.

**Comment by Alkarex** [comment]
The feed works fine for me, but this server seems to perform various types of filtering based on a combination of IP address / User-agent / Accept headers.

Try to put the user-agent of your Web browser in the user-agent field when adding the feed.

![image](https://user-images.githubusercontent.com/1008324/196796520-f8b5362a-781a-49c9-8a42-4b3795e2a3b4.png)

**Reply by svigneau** [reply]
Thanks for the suggestion! However, this doesn't seem to work either.
I forgot to mention that I am using FreshRSS 1.20.0 running on Cloudron.

**Reply by Alkarex** [reply]
There is a risk that it is your server that might be blocked by IP, because it works fine for me.
Do you have access to a terminal from the same server to try a cURL command?

**Reply by svigneau** [reply]
Do you mean running `curl http://nar.oxfordjournals.org/rss/current.xml`? Running it from my server returns:

> <html><body><h1>403 Forbidden</h1>
> Request forbidden by administrative rules.
> </body></html>

Running the same command from my laptop returns the same error, despite being able to open the page on Firefox.

So the issue might be specific to using cURL and not the IP? How does the cURL command work for you?

**Reply by Alkarex** [reply]
```sh
curl -L 'http://nar.oxfordjournals.org/rss/current.xml' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0)' -H 'Accept: application/atom+xml, application/rss+xml, application/rdf+xml;q=0.9, application/xml;q=0.8, text/xml;q=0.8, */*;q=0.1'
```

That one for instance works for me

**Reply by svigneau** [reply]
Oh, thanks for adding the header parameters! Running the same command as you did gives back some javascript code instead of an RSS feed:

> <html><head><script type="text/javascript"><\!--
> function leastFactor(n) {
>  if (isNaN(n) || !isFinite(n)) return NaN;
>  if (typeof phantom !== 'undefined') return 'phantom';
>  if (typeof module !== 'undefined' && module.exports) return 'node';
>  if (n==0) return 0;
>  if (n%1 || n*n<2) return 1;
>  if (n%2==0) return 2;
>  if (n%3==0) return 3;
>  if (n%5==0) return 5;
>  var m=Math.sqrt(n);
>  for (var i=7;i<=m;i+=30) {
>   if (n%i==0)      return i;
>   if (n%(i+4)==0)  return i+4;
>   if (n%(i+6)==0)  return i+6;
>   if (n%(i+10)==0) return i+10;
>   if (n%(i+12)==0) return i+12;
>   if (n%(i+16)==0) return i+16;
>   if (n%(i+22)==0) return i+22;
>   if (n%(i+24)==0) return i+24;
>  }
>  return n;
> }
> function go() {
>  var p=1479898448396; var s=3676587257; var n;
> if ((s >> 11) & 1)/*
> *13;
> */p+=/*
> *13;
> */35719192*/*
> else p-=
> */12; else      p-=/*
> p+= */164567614*12;/*
> *13;
> */if ((s >> 15) & 1)/*
> *13;
> */p+=/*
> *13;
> */88112466*/*
> p+= */16;       else  p-=/*
> p+= */77025078*16; if ((s >> 2) & 1)/*
> p+= */p+=/* 120886108*
> */301381262*5; else     p-=/*
> p+= */313608140*/*
> else p-=
> */3;/* 120886108*
> */if ((s >> 11) & 1)p+=/*
> *13;
> */169869536*/*
> else p-=
> */12;/*
> p+= */else p-=
> 92068119*/*
> else p-=
> */12;   if ((s >> 0) & 1)/*
> else p-=
> */p+=/*
> *13;
> */609240931*/*
> p+= */3;/* 120886108*
> */else 
> p-=324428706*/*
> *13;
> */1;  p+=2282257346;
>  n=leastFactor(p);
> { document.cookie="KEY="+n+"*"+p/n+":"+s+":3994312288:1;path=/; SameSite=none; Secure;";
>   document.location.reload(true); }
> }
> //--></script></head>
> <body onload="go()">
> Loading...
> </body>
> </html>

**Reply by Alkarex** [reply]
If you find a cURL command that works from your server, then the same parameters can be used in FreshRSS. Otherwise you need to use a proxy for that feed, i.e. going through another machine from which it works.

**Reply by svigneau** [reply]
Thanks for your help! I will post here if I find a way that works.

**Reply by svigneau** [reply]
I realized I forgot to mention that I run a FreshRSS install on Cloudron. I also noticed that the curl command fails soon after the RSS server returns `< Cache-Control: no-cache` whereas for a successful call from my laptop, the server returns `< Cache-Control: max-age=31536000`. Cloudron separates code and data with the former not writable. Could this be all related, e.g. the RSS server may need to write cache data, but this is not possible on a Cloudron install?

**Reply by Alkarex** [reply]
If you have other feeds that work, then it is probably not the problem. This HTTP header should not have any impact on the input/output requirements at FreshRSS level.

**Reply by svigneau** [reply]
Ok, thanks for clarifying!
